test/e2e/setup.go

```diff
@@ -73,6 +83,156 @@ func skipIfNotInDevelopmentEnv() {
 	}
 }
 
+func TakeScreenshot(wd WebDriver, e error) {
+	log.Info("Taking Screenshot and saving page source")
+	imageBytes, err := wd.Screenshot()
+	if err != nil {
+		panic(err)
+	}
+
+	imageData, err := png.Decode(bytes.NewReader(imageBytes))
+	if err != nil {
+		panic(err)
+	}
+
+	sourceString, err := wd.PageSource()
+	if err != nil {
+		panic(err)
+	}
+
+	errorString := strings.ReplaceAll(e.Error(), " ", "_")
+
+	imagePath := "./" + errorString + ".png"
+	sourcePath := "./" + errorString + ".html"
+
+	imageAbsPath, err := filepath.Abs(imagePath)
+	if err != nil {
+		panic(err)
+	}
+	sourceAbsPath, err := filepath.Abs(sourcePath)
+	if err != nil {
+		panic(err)
+	}
+
+	image, err := os.Create(imageAbsPath)
+	if err != nil {
+		panic(err)
+	}
+
+	source, err := os.Create(sourceAbsPath)
+	if err != nil {
+		panic(err)
+	}
+
+	err = png.Encode(image, imageData)
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = source.WriteString(sourceString)
+	if err != nil {
+		panic(err)
+	}
+
+	err = image.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	err = source.Close()
+	if err != nil {
+		panic(err)
+	}
+
+	log.Infof("Screenshot saved to %s", imageAbsPath)
+	log.Infof("Page Source saved to %s", sourceAbsPath)
+
+	panic(e)
+}
+
+func adminPortalSessionSetup() (string, *WebDriver) {
+	const (
+		port = 4444
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	os.Setenv("SE_SESSION_REQUEST_TIMEOUT", "9000")
+
+	caps := Capabilities{
+		"browserName":         "MicrosoftEdge",
+		"acceptInsecureCerts": true,
+	}
+	wd := WebDriver(nil)
+
+	var err error
+
+	time.Sleep(time.Second * 10)
+
+	_, err = url.ParseRequestURI("https://localhost:4444")
+	if err != nil {
+		panic(err)
+	}
+
+	for i := 1; i < 5; i++ {
+		wd, err = NewRemote(caps, fmt.Sprintf("http://localhost:%d/wd/hub", port))
+		if wd != nil || ctx.Err() != nil {
+			break
+		}
+		time.Sleep(time.Second * 1)
+	}
+
+	if err != nil {
+		panic(err)
+	}
+
+	log := utillog.GetLogger()
+
+	gob.Register(time.Time{})
+
+	// Navigate to the simple playground interface.
+	host, isNotLocal := os.LookupEnv("PORTAL_HOSTNAME")
+	if !isNotLocal {
```

Comment:
Double negatives are hard.  Essentially what this code is doing (to my understanding) is if a portal hostname is set (in the case of a remote portal) use the env var.  Otherwise default to localhost.  I would do something like this, or sub out `/s/exists/ok` as per normal fashion.  
```suggestion
	host, exists := os.LookupEnv("PORTAL_HOSTNAME")
	if !exists {
```

Reply:
Fixed!